govcd/vapp_network_test.go

```diff
@@ -0,0 +1,250 @@
+	check.Assert(err, IsNil)
+	check.Assert(catalog, NotNil)
+	catalogItem, err := catalog.GetCatalogItemByName(vcd.config.VCD.Catalog.CatalogItem, false)
+	check.Assert(err, IsNil)
+	vappTemplate, err := catalogItem.GetVAppTemplate()
+	check.Assert(err, IsNil)
+
+	vm, err := spawnVM("FirstNode", *vdc, *vapp, desiredNetConfig, vappTemplate, check, false, false)
+	check.Assert(err, IsNil)
+
+	vm2, err := spawnVM("SecondNode", *vdc, *vapp, desiredNetConfig, vappTemplate, check, false, false)
+	check.Assert(err, IsNil)
+
+	uuid, err := GetUuidFromHref(networkFound.Link.HREF, false)
+	check.Assert(err, IsNil)
+
+	result, err := vapp.UpdateNetworkNatRules(uuid, []*types.NatRule{&types.NatRule{VMRule: &types.NatVMRule{
+		ExternalPort: -1, InternalPort: 22, VMNicID: 0,
+		VAppScopedVMID: vm.VM.VAppScopedLocalID, Protocol: "TCP"}},
+		&types.NatRule{VMRule: &types.NatVMRule{
+			ExternalPort: 80, InternalPort: 22, VMNicID: 0,
+			VAppScopedVMID: vm2.VM.VAppScopedLocalID, Protocol: "UDP"}}},
+		"portForwarding", "allowTraffic")
+	check.Assert(err, IsNil)
+	check.Assert(result, NotNil)
+
+	// verify
+	check.Assert(result.Configuration.Features.NatService.Policy, Equals, "allowTraffic")
+	check.Assert(result.Configuration.Features.NatService.NatType, Equals, "portForwarding")
+
+	check.Assert(result.Configuration.Features.NatService.NatRule[0].VMRule.InternalPort, Equals, 22)
+	check.Assert(result.Configuration.Features.NatService.NatRule[0].VMRule.ExternalPort, Equals, -1)
+	check.Assert(result.Configuration.Features.NatService.NatRule[0].VMRule.VAppScopedVMID, Equals, vm.VM.VAppScopedLocalID)
+	check.Assert(result.Configuration.Features.NatService.NatRule[0].VMRule.VMNicID, Equals, 0)
+
+	check.Assert(result.Configuration.Features.NatService.NatRule[1].VMRule.InternalPort, Equals, 22)
+	check.Assert(result.Configuration.Features.NatService.NatRule[1].VMRule.ExternalPort, Equals, 80)
+	check.Assert(result.Configuration.Features.NatService.NatRule[1].VMRule.VAppScopedVMID, Equals, vm2.VM.VAppScopedLocalID)
+	check.Assert(result.Configuration.Features.NatService.NatRule[1].VMRule.VMNicID, Equals, 0)
+
+	result, err = vapp.UpdateNetworkNatRules(uuid, []*types.NatRule{&types.NatRule{OneToOneVMRule: &types.NatOneToOneVMRule{
+		MappingMode: "automatic", VMNicID: 0,
+		VAppScopedVMID: vm.VM.VAppScopedLocalID}},
+		&types.NatRule{OneToOneVMRule: &types.NatOneToOneVMRule{
+			MappingMode: "manual", VMNicID: 0,
+			VAppScopedVMID:    vm2.VM.VAppScopedLocalID,
+			ExternalIPAddress: takeStringPointer("192.168.100.1")}}},
+		"ipTranslation", "allowTrafficIn")
+	check.Assert(err, IsNil)
+	check.Assert(result, NotNil)
+
+	// verify
+	check.Assert(result.Configuration.Features.NatService.Policy, Equals, "allowTrafficIn")
+	check.Assert(result.Configuration.Features.NatService.NatType, Equals, "ipTranslation")
+
+	check.Assert(result.Configuration.Features.NatService.NatRule[0].OneToOneVMRule.MappingMode, Equals, "automatic")
```

Comment:
Check that `result.Configuration.Features.NatService.NatRule` has at least 2 elements